modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/enricher/ClassLoaderModelEnricher.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.introspection.enricher;
+
+import org.mule.runtime.extension.api.introspection.ExtensionModel;
+import org.mule.runtime.extension.api.introspection.declaration.DescribingContext;
+import org.mule.runtime.extension.api.introspection.declaration.spi.ModelEnricher;
+import org.mule.runtime.extension.api.introspection.property.ClassLoaderModelProperty;
+
+/**
+ * Adds a {@link ClassLoaderModelProperty} at the {@link ExtensionModel} level, which
```

Comment:
I don't know if there is currently a difference in the implementation, but the classloader must be the plugin's classloader, which could not necessarily be the one on the thread's context